# 1. 필요 csv 불러오기
"금호동_최종_데이터.csv" 데이터 불러오기

In [1]:
import pandas as pd
df = pd.read_csv('금호동_최종_데이터.csv',index_col=0)
df

gid  상권개수   인구정보      연면적  Busdistance     HubDist    trash  \
0     라라212603     0  154.0  1083.21   144.183573   13.308311  12900.0   
1     라라199601     0  138.0  2261.98    61.306397   30.667224      0.0   
2     라라209607     0   77.0  1023.00   115.206453   36.987092      0.0   
3     라라207605     0    9.0   676.50   163.932305    9.244004   4000.0   
4     라라201601     0  132.0  2135.38   215.299204   17.696178      0.0   
...        ...   ...    ...      ...          ...         ...      ...   
2282  라라237581     0    0.0     0.00  2019.788254  404.705986      0.0   
2283  라라223572     0    0.0     0.00  2451.218420  217.809801      0.0   
2284  라라190581     0    0.0     0.00  1789.629515  116.939443      0.0   
2285  라라248553     0    0.0     0.00  4788.218514  214.805182      0.0   
2286  라라243554     0    0.0     0.00  4655.003596  248.116508      0.0   

                                               geometry  
0     MULTIPOLYGON (((127.732147404193 34.9373046468...  
1     MULTIPOLYGON (((127.7179072746085 34.935527519...  
2     MULTIPOLYGON (((127.7288723433386 34.940917863...  
3     MULTIPOLYGON (((127.7266772301307 34.939118477...  
4     MULTIPOLYGON (((127.7200972832025 34.935523571...  
...                                                 ...  
2282  MULTIPOLYGON (((127.7594605813311 34.917410847...  
2283  MULTIPOLYGON (((127.7441098973286 34.909326865...  
2284  MULTIPOLYGON (((127.7080067112277 34.917509812...  
2285  MULTIPOLYGON (((127.7714198466552 34.892135590...  
2286  MULTIPOLYGON (((127.7659506311317 34.893049435...  

[2287 rows x 8 columns]

# 2. 모델링 전 상권개수, 인구정보, 연면적, trash 변수 값이 모두 0인 값인 raw 삭제

In [2]:
a = df[(df['상권개수']==0) & (df['인구정보']==0) & (df['연면적']==0) & (df['trash']==0)]
idx = a.index
idx

Int64Index([  94,   95,   96,   97,   98,   99,  100,  101,  102,  103,
            ...
            2277, 2278, 2279, 2280, 2281, 2282, 2283, 2284, 2285, 2286],
           dtype='int64', length=2123)

In [3]:
df=df.drop(idx).reset_index(drop=True)

In [4]:
df.shape

(164, 8)

In [5]:
df

gid  상권개수   인구정보      연면적  Busdistance    HubDist    trash  \
0    라라212603     0  154.0  1083.21   144.183573  13.308311  12900.0   
1    라라199601     0  138.0  2261.98    61.306397  30.667224      0.0   
2    라라209607     0   77.0  1023.00   115.206453  36.987092      0.0   
3    라라207605     0    9.0   676.50   163.932305   9.244004   4000.0   
4    라라201601     0  132.0  2135.38   215.299204  17.696178      0.0   
..        ...   ...    ...      ...          ...        ...      ...   
159  라라240554     0    0.0   192.88  4593.398993  24.072767      0.0   
160  라라207604     0    0.0   194.40   129.361336  49.408544      0.0   
161  라라197603     2    0.0   201.22   315.437198  35.252876      0.0   
162  라라212599    10    0.0  1732.48   268.246452  27.638026      0.0   
163  라라242578     0    0.0   374.40  2248.465194  44.407827      0.0   

                                              geometry  
0    MULTIPOLYGON (((127.732147404193 34.9373046468...  
1    MULTIPOLYGON (((127.7179072746085 34.935527519...  
2    MULTIPOLYGON (((127.7288723433386 34.940917863...  
3    MULTIPOLYGON (((127.7266772301307 34.939118477...  
4    MULTIPOLYGON (((127.7200972832025 34.935523571...  
..                                                 ...  
159  MULTIPOLYGON (((127.7626673207969 34.893056574...  
160  MULTIPOLYGON (((127.7266747490325 34.938216731...  
161  MULTIPOLYGON (((127.7157219875493 34.937334922...  
162  MULTIPOLYGON (((127.7321372416677 34.933697661...  
163  MULTIPOLYGON (((127.7649256897607 34.914693796...  

[164 rows x 8 columns]

# 3. 모델링

## 3-1. 데이터 정규화
- MinMaxScaler

In [6]:
# train_test_split
from sklearn.model_selection import train_test_split
X_train, X_test, y_train,y_test = train_test_split(df.iloc[:,1:6], df.iloc[:,6], random_state=1234, test_size= 0.3)
print(X_train.shape,X_test.shape)

(114, 5) (50, 5)


In [7]:
from sklearn.preprocessing import MinMaxScaler
minmaxScaler = MinMaxScaler()
minmaxScaler.fit(X_train)

X_train_scaled = minmaxScaler.transform(X_train)
X_test_scaled = minmaxScaler.transform(X_test)

## 3-2. 데이터 모델링

In [12]:
from sklearn.svm import LinearSVR
linear_svr = LinearSVR(random_state=1234).fit(X_train_scaled,y_train)

In [13]:
linear_svr.coef_

array([-0.320905  ,  2.36554888,  0.04822848,  0.09531933, -0.62530756])

In [14]:
linear_svr.intercept_

array([-0.59462187])

# 4. 최종 결과

변수별 가중치 값 최종

  상권개수	     인구정보	  연면적	    Busdistance	 HubDist	
  
: [-0.320905  ,  2.36554888,  0.04822848,  0.09531933, -0.62530756]

y 절편 : [-0.59462187]